.github/workflows/functional-tests.yml

```
@@ -0,0 +1,53 @@
+---
+name: functional-tests
+
+on:
+  workflow_dispatch: ~
+  schedule:
+    - cron: '0 3 * * 1-5'
+
+permissions:
+  contents: read
+  id-token: write
+
```

Can you add the TF variables:


```suggestion

env:
    TF_VAR_BRANCH: ${{ github.ref_name }}
    TF_VAR_BUILD_ID: ${{ github.run_id }}
    TF_VAR_ENVIRONMENT: 'ci'
    TF_VAR_REPO: ${{ github.repository }}

```